molecule/app_deploy_domain/prepare.yml

```diff
@@ -0,0 +1,198 @@
+---
+- name: Prepare
+  hosts: all
+  gather_facts: true
+  vars_files:
+    - ../group_vars/all/vars.yml
+  vars:
+    sudo_pkg_name: 'sudo'
+  tasks:
+    - name: "Ensure {{ sudo_pkg_name }} is installed (if user is root)."
+      ansible.builtin.yum:
+        name: "{{ sudo_pkg_name }}"
+      when:
+        - ansible_user_id == 'root'
+
+    - name: Gather the package facts
+      ansible.builtin.package_facts:
+        manager: auto
+
+    - name: "Check if {{ sudo_pkg_name }} is installed."
+      ansible.builtin.assert:
+        that:
+          - sudo_pkg_name in ansible_facts.packages
+
+    - name: "Ensure required packages are installed."
+      become: "{{ wildfly_install_requires_become | default(true) }}"
+      ansible.builtin.yum:
+        name:
+          - java-17-openjdk
+          - procps-ng
+          - tzdata-java
+        state: present
+
+    - name: "Ensure files directory exists on controller"
+      ansible.builtin.file:
+        path: "{{ playbook_dir }}/../files"
+        state: directory
+        mode: '0755'
+      delegate_to: localhost
+      run_once: true
+
+    - name: "Download test application to controller"
+      ansible.builtin.get_url:
+        url: "https://github.com/rpelisse/simple-servlet/releases/download/info_wfly31_1.2/info-1.2.war"
+        dest: "{{ playbook_dir }}/../files/test-app.war"
+        mode: '0644'
+      delegate_to: localhost
+      run_once: true
+
+    - name: "Install WildFly"
+      ansible.builtin.include_role:
+        name: wildfly_install
+      # Note: wildfly_install always uses standalone/configuration/, 
+      # so we'll set up domain mode manually after installation
+
+    - name: "Set wildfly_home fact"
+      ansible.builtin.set_fact:
+        wildfly_home: "{{ wildfly_install_workdir | default('/opt/wildfly/') }}wildfly-{{ wildfly_version | default('38.0.0.Final') }}/"
+      vars:
+        wildfly_install_workdir: '/opt/wildfly/'
+        wildfly_version: '38.0.0.Final'
+
+    - name: "Ensure PID directory exists"
+      ansible.builtin.file:
+        path: /run/wildfly
+        state: directory
+        owner: wildfly
+        group: wildfly
+        mode: '0755'
+      become: true
+
+    - name: "Create domain mode systemd service file"
```

Comment:
Wouldn't it be better use WildFly's official systemd service? https://docs.wildfly.org/36/Getting_Started_Guide.html#installation_as_a_linux_service

Reply:
I will look into Getting started guide.

Reply:
@iweiss @RanabirChakraborty @sabre1041 I was looking into the collection docs, I see we have mentioned we don't support the domain made, I was thinking in that case we should skip this molecule test scenatio and keep only the standalone one. Let me know your taughts.